src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('the side should be bigger than 0');
```

Comment:
The error message here could be more descriptive. For instance, it could specify which side is invalid and that side lengths must be positive numbers.